### PR TITLE
feat: Credential Management

### PR DESCRIPTION
**Pull request by mishasizov-SK**
Given PR is based on https://github.com/trustbloc/vcs/pull/788 with the next additions:
- Unit tests for Issue Credential
- Unit tests for Verify Credential
- BDD tests - show flows for JSON-LD (issue and verify).

More specifically, this PR implements:
- Issue Credential + unit tests.
- Verify Credential + unit tests.
- BDD tests - show flows for JSON-LD (issue and verify).
form https://github.com/trustbloc/vcs/issues/757

Signed-off-by: Mykhailo Sizov <mykhailo.sizov@securekey.com>

### CODE REVIEW

File: pkg/restapi/v1/issuer/controller_test.go [hunk 1]
```diff
@@ -944,3 +1225,138 @@ func TestController_PostIssuerProfilesProfileIDDeactivate(t *testing.T) {
...
+	type args struct {
+		options *IssueCredentialOptions
+	}
+	tests := []struct {
+		name    string
+		args    args
+		want    []crypto.SigningOpts
+		wantErr bool
+	}{
+		// TODO: Add test cases.
```
Comment (aholovko) [inline pkg/restapi/v1/issuer/controller_test.go:1239]: is it going to be added in follow-up tickets? do we have the one created?

File: pkg/restapi/v1/verifier/models.go [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package verifier
+
+import "github.com/trustbloc/vcs/pkg/service/verifycredential"
+
+// CredentialsVerificationResponse represents the response of credentials verification.
+type CredentialsVerificationResponse struct {
```
Comment (vkubiv) [inline pkg/restapi/v1/verifier/models.go:12]: I think this should be generated from openapi.yaml. You can define there this structure

File: pkg/restapi/v1/verifier/controller.go [hunk 3]
```diff
@@ -180,26 +229,65 @@ func (c *Controller) PostVerifierProfilesProfileIDDeactivate(ctx echo.Context, p
...
 		return nil, resterr.NewSystemError(verifierProfileSvcComponent, "GetProfile", err)
 	}
 
-	// block access to profiles of other organizations
-	if profile.OrganizationID != orgID {
+	// Profiles of other organization is not visible.
+	if profile.OrganizationID != oidcOrgID {
 		return nil, resterr.NewValidationError(resterr.DoesntExist, "profile",
-			fmt.Errorf("no profile with id %s", profileID))
+			fmt.Errorf("profile with given id %s, doesn't exists", profileID))
```
Comment (aholovko) [inline pkg/restapi/v1/verifier/controller.go:290]: doesn't exists -> doesn't exist :)

File: pkg/service/credentialstatus/credentialstatus_service.go [hunk 4]
```diff
@@ -0,0 +1,117 @@
...
+	"context"
+	"crypto/tls"
+	"fmt"
+	"io"
+	"net/http"
+
+	"github.com/hyperledger/aries-framework-go/pkg/doc/verifiable"
+	vdrapi "github.com/hyperledger/aries-framework-go/pkg/framework/aries/api/vdr"
+	"github.com/piprate/json-gold/ld"
+
```
Comment (aholovko) [inline pkg/service/credentialstatus/credentialstatus_service.go:19]: remove line here

File: pkg/service/verifycredential/verifycredential_service.go [hunk 5]
```diff
@@ -0,0 +1,313 @@
...
+func (s *Service) VerifyCredential(credential *verifiable.Credential, opts *Options,
+	profile *verifier.Profile) ([]CredentialsVerificationCheckResult, error) {
+	checks := profile.Checks.Credential
+
+	var result []CredentialsVerificationCheckResult
+
+	if checks.Proof {
+		vcBytes, err := json.Marshal(credential)
+		if err != nil {
+			return nil, fmt.Errorf("unexepcted error on credential marshal: %w", err)
```
Comment (aholovko) [inline pkg/service/verifycredential/verifycredential_service.go:91]: unexepcted -> unexpected

**Review by aholovko — APPROVED** [review]

**Review by vkubiv — APPROVED** [review]

**Review by vkubiv — CHANGES_REQUESTED** [review]
Consider to add few additional tests for arieskms.go  , v1/issuer/controller.go, and vc/validate.gopkg/doc/vc/crypto/crypto.go

File: pkg/restapi/v0.1/holder/operation/operations.go [hunk 6]
```diff
@@ -317,7 +331,7 @@ func (o *Operation) deriveCredentialsHandler(rw http.ResponseWriter, req *http.R
 	vcBytes, err := derived.MarshalJSON()
 	if err != nil {
 		commhttp.WriteErrorResponse(rw, http.StatusInternalServerError,
-			fmt.Sprintf("failed to matshal derived vc: %s", err.Error()))
+			fmt.Sprintf("failed to matshal derived vcContent: %s", err.Error()))
```
[the remark this answers is not in the document]
Comment (mishasizov-SK) [reply pkg/restapi/v0.1/holder/operation/operations.go:334]: done.

File: pkg/restapi/v1/issuer/controller_test.go (hunk 1 above)
Comment (mishasizov-SK) [reply pkg/restapi/v1/issuer/controller_test.go:1239]: I totally spaced it out. Added.

File: pkg/restapi/v1/verifier/controller.go (hunk 3 above)
Comment (mishasizov-SK) [reply pkg/restapi/v1/verifier/controller.go:290]: sure, udpated

File: pkg/service/credentialstatus/credentialstatus_service.go (hunk 4 above)
Comment (mishasizov-SK) [reply pkg/service/credentialstatus/credentialstatus_service.go:19]: sure

File: pkg/service/verifycredential/verifycredential_service.go (hunk 5 above)
Comment (mishasizov-SK) [reply pkg/service/verifycredential/verifycredential_service.go:91]: correct

File: pkg/restapi/v1/verifier/models.go (hunk 2 above)
Comment (mishasizov-SK) [reply pkg/restapi/v1/verifier/models.go:12]: added

**Review by vkubiv — APPROVED** [review]

**Pull request merged**